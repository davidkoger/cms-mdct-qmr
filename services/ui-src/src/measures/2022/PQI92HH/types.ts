import * as Types from "measures/2022/CommonQuestions/types";

export type FormData = Types.DefaultFormData;
