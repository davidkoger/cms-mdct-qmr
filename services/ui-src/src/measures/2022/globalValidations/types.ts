import { OmsNodes as OMS } from "measures/2022/CommonQuestions/types";

export interface FormRateField {
  denominator?: string;
  numerator?: string;
  label?: string;
  rate?: string;
  isTotal?: string;
}

export type locationDictionaryFunction = (labels: string[]) => string;

export interface RateData extends OMS.OmsRateFields {
  "pcr-rate"?: { id?: number; value?: string; label?: string }[];
  "iuhh-rate"?: any;
  "aifhh-rate"?: any;
}

export interface UnifiedValFuncProps {
  categories?: string[];
  qualifiers?: string[];
  rateData: FormRateField[][];
  location: string;
  errorMessage?: string;
}

export type UnifiedValidationFunction = (
  props: UnifiedValFuncProps
) => FormError[];

export type OmsValidationCallback = (data: {
  rateData: RateData;
  qualifiers: string[];
  categories: string[];
  label: string[];
  locationDictionary: locationDictionaryFunction;
  isOPM: boolean;
  customTotalLabel?: string;
  dataSource?: string[];
}) => FormError[];
