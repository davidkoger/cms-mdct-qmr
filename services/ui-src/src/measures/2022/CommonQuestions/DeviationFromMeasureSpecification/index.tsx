import * as QMR from "components";
import * as Types from "../types";
import * as DC from "dataConstants";
import { useWatch } from "react-hook-form";
import { useCustomRegister } from "hooks/useCustomRegister";

interface GetTopLvlDeviationOptions {
  categories: string[];
  customTotalLabel?: string;
}

type TopLevelOptions = {
  displayValue: string;
  rates: Types.RateFields[] | undefined;
  key: string;
}[];

interface Props {
  categories: string[];
  customTotalLabel?: string;
  measureName?: string;
}

interface OptionProps {
  name: string;
  qualifiers?: Types.RateFields[];
}

/**
 * It removes all non-word characters from a string.
 * @param {string} [s] - The string to be cleaned.
 */
const cleanString = (s?: string) => s?.replace(/[^\w]/g, "");

/**
 * Check if the rate has both a numerator and a denominator.
 * @param el - The rate field that we're checking.
 */
const numDenExistInRate = (el: Types.RateFields) =>
  el?.numerator && el?.denominator;

/**
 * Check if the rates within a qualifier have both a numerator and a denominator.
 * @param qualifier - The qualifier that we're checking.
 */
const complexNumDenExistInRate = (qualifier: any) => {
  return qualifier.fields.some(
    (field: any) => field?.value && field?.value !== ""
  );
};

/**
 * It returns an array of objects that contain a display value, value, and single TextArea child
 * @param {string} name - The name of the field.
 */
const getRateTextAreaOptions = (name: string) =>
  ["Numerator", "Denominator", "Other"].map((displayValue) => {
    const value = displayValue.toLowerCase();
    return {
      displayValue,
      value,
      children: [
        <QMR.TextArea
          label="Explain:"
          name={`${name}.${value}`}
          key={`${name}.${value}`}
        />,
      ],
    };
  });

/* This is a custom checkbox component that is used to render the checkboxes for the rate deviations. */
const DeviationsSelectedCheckbox = ({ name }: { name: string }) => (
  <QMR.Checkbox
    name={`${name}.${DC.RATE_DEVIATIONS_SELECTED}`}
    key={`${name}.${DC.RATE_DEVIATIONS_SELECTED}`}
    options={getRateTextAreaOptions(name)}
  />
);

/**
 * It takes in a list of qualifiers and returns options based on if the rates contain a label
 * @param {OptionProps}  - qualifier: the qualifier object - name: a name to register the input with
 * @returns A list of options
 */
export const getLowLvlDeviationOptions = ({
  qualifiers,
  name,
}: OptionProps) => {
  if (!qualifiers || qualifiers.length === 0) return [];

  // if there are no labels then there is no need for the additional checkbox
  if (!qualifiers.some((el) => el.label)) {
    return getRateTextAreaOptions(name);
  }

  return qualifiers
    .sort((a, b) => (a.label!! < b.label!! ? 1 : 1))
    .map((item) => {
      const value = `${cleanString(item.label)}`;
      return {
        displayValue: item.label!,
        value,
        children: [
          <DeviationsSelectedCheckbox
            name={`${name}.${value}`}
            key={`${name}.${value}`}
          />,
        ],
      };
    });
};

export const PCRADgetLowLvlDeviationOptions = ({
  qualifiers,
  name,
}: OptionProps) => {
  if (!qualifiers || qualifiers.length === 0) return [];

  return getRateTextAreaOptions(name);
};

export const DeviationFromMeasureSpec = ({
  categories,
  measureName = "",
  customTotalLabel,
}: Props) => {
  const register = useCustomRegister<Types.DeviationFromMeasureSpecification>();
  const watchPerformanceMeasure = useWatch({
    name: DC.PERFORMANCE_MEASURE,
  });

  const getTopLvlDeviationOptions = ({
    categories,
    customTotalLabel,
  }: GetTopLvlDeviationOptions) => {
    if (watchPerformanceMeasure?.rates) {
      const topLvlOptions: TopLevelOptions = [];
      const { rates } = watchPerformanceMeasure;

      if (rates.singleCategory) {
        // handle for PCR-XX measures
        if (["PCR-AD", "PCR-HH"].includes(measureName)) {
          const quals = rates.singleCategory.filter((r: any) => r.value !== "");
          if (quals.length > 0) {
            return getRateTextAreaOptions(DC.DEVIATIONS);
          }
        }
        // A total category should have the label "Total", per the Figma design.
        const totalIndex = rates.singleCategory.findIndex(
          (cat: any) => cat.isTotal === true
        );
        if (totalIndex >= 0) {
          rates.singleCategory[totalIndex].label = `${
            customTotalLabel ? `${customTotalLabel} ` : "Total"
          }`;
        }
        /* This is checking if the rates object has a singleCategory key.
        If it does, then it will return the low level deviation options. */
        return getLowLvlDeviationOptions({
          qualifiers:
            measureName === "AIF-HH"
              ? rates.singleCategory.filter(complexNumDenExistInRate)
              : rates.singleCategory.filter(numDenExistInRate),
          name: DC.DEVIATIONS,
        });
      } else {
        categories.forEach((cat) => {
          const key = cat.replace(/[^\w]/g, "");
          // if some of the rates have both num and den
          const deviations =
            measureName === "IU-HH"
              ? rates[key]?.some(complexNumDenExistInRate)
              : rates[key]?.some(numDenExistInRate);
          const deviationRates =
            measureName === "IU-HH"
              ? rates[key]?.filter(complexNumDenExistInRate)
              : rates[key]?.filter(numDenExistInRate);

          if (deviations) {
            // add the rates that have num and den to topLvlOptions along with its display value from categories
            topLvlOptions.push({
              rates: deviationRates,
              displayValue: cat,
              key,
            });
          }
        });
      }

      return (
        topLvlOptions?.map((option) => {
          return {
            value: option.key,
            displayValue: option.displayValue,
            children: [
              <QMR.Checkbox
                {...register(
                  `${DC.DEVIATIONS}.${option.key}.${DC.SELECTED_OPTIONS}`
                )}
                formLabelProps={{ fontWeight: 600 }}
                key={`${DC.DEVIATIONS}.${option.key}`}
                options={getLowLvlDeviationOptions({
                  qualifiers: option.rates,
                  name: `${DC.DEVIATIONS}.${option.key}`,
                })}
              />,
            ],
          };
        }) ?? []
      );
    }
    return [];
  };

  return (
    <QMR.CoreQuestionWrapper label="Deviations from Measure Specifications">
      <QMR.RadioButton
        renderHelperTextAbove
        {...register(DC.DID_CALCS_DEVIATE)}
        formLabelProps={{ fontWeight: 600 }}
        label="Did your calculation of the measure deviate from the measure specification in any way?"
        helperText="For example: deviation from measure specification might include different methodology, timeframe, or reported age groups."
        options={[
          {
            displayValue:
              "Yes, the calculation of the measure deviates from the measure specification.",
            value: DC.YES,
            children: [
              <QMR.Checkbox
                {...register(DC.DEVIATION_OPTIONS)}
                label="Select and explain the deviation(s):"
                options={getTopLvlDeviationOptions({
                  categories,
                  customTotalLabel,
                })}
              />,
            ],
          },
          {
            displayValue:
              "No, the calculation of the measure does not deviate from the measure specification in any way.",
            value: DC.NO,
          },
        ]}
      />
    </QMR.CoreQuestionWrapper>
  );
};
