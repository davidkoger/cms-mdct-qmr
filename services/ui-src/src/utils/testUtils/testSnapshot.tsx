import { renderWithHookForm } from "./reactHookFormRenderer";
import { useApiMock } from "utils/testUtils/useApiMock";

interface AdditionalOptions {
  defaultValues?: Object;
}

// interface APIData {}

interface Props {
  component: JSX.Element;
  defaultValues?: AdditionalOptions["defaultValues"];
  apiData?: any;
}
/**
 * Compare a component with a stored snapshot.
 * @param {JSX.Element} component Component to be compared with snapshot
 * @param {AdditionalOptions} defaultValues ReactHookForm data to be passed to the render function
 */
export const testSnapshot = ({
  component,
  defaultValues = {},
  apiData = {},
}: Props) => {
  useApiMock(apiData);
  const { container } = renderWithHookForm(component, defaultValues);
  cleanAttributes(container);
  expect(container).toMatchSnapshot();
};

/**
 * Clean up attributes from the rendered component by removing the ids generated by Chakra.
 *
 * Before:
 *   for="field-1379718737-40"
 *   id="field-1379718737-40-label"
 *
 * After:
 *   for="field-40"
 *   id="field-40-label"
 *
 * This is useful during snapshot testing. Snapshot comparisons will fail without addressing these items.
 *
 * @param {HTMLElement} container The component to clean up.
 */
const cleanAttributes = (container: HTMLElement) => {
  const attributes = [
    "id",
    "for",
    "aria-controls",
    "aria-describedby",
    "aria-labelledby",
  ];
  const prefixes = ["field", "radio", "popover"];

  const clean = (attr: string) => {
    const attrList = attr.split("-");
    // I can't find a better way to do this in the Chakra docs. If there is though, we should use it.
    if (prefixes.includes(attrList[0])) {
      if (attrList[0] === "popover") {
        attrList.splice(2, 1);
      } else {
        attrList.splice(1, 1);
      }
      return attrList.join("-");
    } else {
      return attr;
    }
  };

  attributes.forEach((attr) => {
    if (container.hasAttribute(attr)) {
      const value = clean(container.getAttribute(attr)!);
      container.setAttribute(attr, value);
    }
  });

  Array.from(container.children).forEach((child) => {
    cleanAttributes(child as HTMLElement);
  });
};
