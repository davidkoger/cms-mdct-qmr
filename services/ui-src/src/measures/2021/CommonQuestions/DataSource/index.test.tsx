import { DataSource } from "./index";
import { DataSourceData as DS } from "utils/testUtils/testFormData";
import { DataDrivenTypes } from "measures/2021/CommonQuestions/types";
import { testSnapshot } from "utils/testUtils/testSnapshot";

describe("Test the global DataSource component", () => {
  it("(Default) Component renders with correct content", () => {
    const component = <DataSource />;
    testSnapshot({ component, defaultValues: { defaultValues: DS.default } });
  });

  it("(Custom Structure) Component renders with correct content", () => {
    const component = <DataSource data={data} />;
    testSnapshot({ component, defaultValues: { defaultValues: DS.custom } });
  });
});

const data: DataDrivenTypes.DataSource = {
  optionsLabel: "How do we feel about a label with this text? Does it work?",
  options: [
    {
      value: "Option 1",
      subOptions: [
        {
          label: "What is the Option 1 Data Source?",
          options: [
            {
              value: "I'm not telling",
            },
            {
              value: "You'll have to kill me first",
              description: true,
            },
          ],
        },
      ],
    },
    {
      value: "A Better Option",
      description: true,
    },
    {
      value: "Something Even Crazier Than The First 2",
      description: false,
    },
  ],
};
