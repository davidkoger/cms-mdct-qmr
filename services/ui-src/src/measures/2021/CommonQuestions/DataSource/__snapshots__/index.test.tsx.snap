// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the global DataSource component (Custom Structure) Component renders with correct content 1`] = `
<body>
  <div
    class="css-pb69ky"
  >
    <h2
      class="chakra-form__label css-e3y4kl"
      data-cy="Data Source"
    >
      Data Source
    </h2>
    <fieldset>
      <div
        data-cy="data-source-options"
      >
        <div
          class="chakra-form-control css-0"
          role="group"
        >
          <label
            class="chakra-form__label css-uvho5g"
            data-cy="How do we feel about a label with this text? Does it work?"
            for="field-6"
            id="field-6-label"
          >
            How do we feel about a label with this text? Does it work?
          </label>
          <div
            class="chakra-stack css-n21gh5"
          >
            <div
              class="css-79elbk"
            >
              <label
                class="chakra-checkbox css-1uiwwan"
                data-checked=""
                data-cy="DataSource0"
              >
                <input
                  checked=""
                  class="chakra-checkbox__input"
                  id="DataSource0-checkbox"
                  style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
                  type="checkbox"
                  value="Option1"
                />
                <span
                  aria-hidden="true"
                  class="chakra-checkbox__control css-dnty2r"
                  data-checked=""
                >
                  <div
                    style="display: flex; align-items: center; justify-content: center; height: 100%; transform: none;"
                  >
                    <svg
                      class="css-o8vbdw"
                      style="fill: none; stroke-width: 2; stroke: currentColor; stroke-dasharray: 16; opacity: 1; stroke-dashoffset: 0;"
                      viewBox="0 0 12 10"
                    >
                      <polyline
                        points="1.5 6 4.5 9 10.5 1"
                      />
                    </svg>
                  </div>
                </span>
                <span
                  class="chakra-checkbox__label css-1oeb2oe"
                  data-checked=""
                >
                  <p
                    class="chakra-text css-vjer7o"
                  >
                    Option 1
                  </p>
                </span>
              </label>
              <div
                class="chakra-collapse"
                style="overflow: initial; display: block; opacity: 1; height: auto;"
              >
                <div
                  class="css-1s47xlp"
                >
                  <div
                    class="chakra-form-control css-0"
                    role="group"
                  >
                    <label
                      class="chakra-form__label css-uvho5g"
                      data-cy="What is the Option 1 Data Source?"
                      for="field-7"
                      id="field-7-label"
                    >
                      What is the Option 1 Data Source?
                    </label>
                    <div
                      class="chakra-stack css-n21gh5"
                    >
                      <div
                        class="css-79elbk"
                      >
                        <label
                          class="chakra-checkbox css-1uiwwan"
                          data-checked=""
                          data-cy="DataSourceSelections.Option10.selected0"
                        >
                          <input
                            checked=""
                            class="chakra-checkbox__input"
                            id="DataSourceSelections.Option10.selected0-checkbox"
                            style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
                            type="checkbox"
                            value="Imnottelling"
                          />
                          <span
                            aria-hidden="true"
                            class="chakra-checkbox__control css-dnty2r"
                            data-checked=""
                          >
                            <div
                              style="display: flex; align-items: center; justify-content: center; height: 100%; transform: none;"
                            >
                              <svg
                                class="css-o8vbdw"
                                style="fill: none; stroke-width: 2; stroke: currentColor; stroke-dasharray: 16; opacity: 1; stroke-dashoffset: 0;"
                                viewBox="0 0 12 10"
                              >
                                <polyline
                                  points="1.5 6 4.5 9 10.5 1"
                                />
                              </svg>
                            </div>
                          </span>
                          <span
                            class="chakra-checkbox__label css-1oeb2oe"
                            data-checked=""
                          >
                            <p
                              class="chakra-text css-vjer7o"
                            >
                              I'm not telling
                            </p>
                          </span>
                        </label>
                        <div
                          class="chakra-collapse"
                          style="overflow: initial; display: block; opacity: 1; height: auto;"
                        >
                          <div
                            class="css-1ysiprv"
                          />
                        </div>
                      </div>
                      <div
                        class="css-79elbk"
                      >
                        <label
                          class="chakra-checkbox css-1uiwwan"
                          data-checked=""
                          data-cy="DataSourceSelections.Option10.selected1"
                        >
                          <input
                            checked=""
                            class="chakra-checkbox__input"
                            id="DataSourceSelections.Option10.selected1-checkbox"
                            style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
                            type="checkbox"
                            value="Youllhavetokillmefirst"
                          />
                          <span
                            aria-hidden="true"
                            class="chakra-checkbox__control css-dnty2r"
                            data-checked=""
                          >
                            <div
                              style="display: flex; align-items: center; justify-content: center; height: 100%; transform: none;"
                            >
                              <svg
                                class="css-o8vbdw"
                                style="fill: none; stroke-width: 2; stroke: currentColor; stroke-dasharray: 16; opacity: 1; stroke-dashoffset: 0;"
                                viewBox="0 0 12 10"
                              >
                                <polyline
                                  points="1.5 6 4.5 9 10.5 1"
                                />
                              </svg>
                            </div>
                          </span>
                          <span
                            class="chakra-checkbox__label css-1oeb2oe"
                            data-checked=""
                          >
                            <p
                              class="chakra-text css-vjer7o"
                            >
                              You'll have to kill me first
                            </p>
                          </span>
                        </label>
                        <div
                          class="chakra-collapse"
                          style="overflow: initial; display: block; opacity: 1; height: auto;"
                        >
                          <div
                            class="css-1s47xlp"
                          >
                            <div
                              class="chakra-form-control css-0"
                              role="group"
                            >
                              <label
                                class="chakra-form__label css-uvho5g"
                                data-cy="Describe the data source:"
                                for="field-8"
                                id="field-8-label"
                              >
                                Describe the data source:
                              </label>
                              <textarea
                                class="chakra-textarea css-0"
                                data-cy="DataSourceSelections.Option10-Youllhavetokillmefirst.description"
                                id="field-8"
                                name="DataSourceSelections.Option10-Youllhavetokillmefirst.description"
                              >
                                Another one bites the dust
                              </textarea>
                              <div
                                class="css-k008qs"
                              >
                                <div
                                  class="css-17xejub"
                                />
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                    <div
                      class="css-k008qs"
                    >
                      <div
                        class="css-17xejub"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="css-79elbk"
            >
              <label
                class="chakra-checkbox css-1uiwwan"
                data-checked=""
                data-cy="DataSource1"
              >
                <input
                  checked=""
                  class="chakra-checkbox__input"
                  id="DataSource1-checkbox"
                  style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
                  type="checkbox"
                  value="ABetterOption"
                />
                <span
                  aria-hidden="true"
                  class="chakra-checkbox__control css-dnty2r"
                  data-checked=""
                >
                  <div
                    style="display: flex; align-items: center; justify-content: center; height: 100%; transform: none;"
                  >
                    <svg
                      class="css-o8vbdw"
                      style="fill: none; stroke-width: 2; stroke: currentColor; stroke-dasharray: 16; opacity: 1; stroke-dashoffset: 0;"
                      viewBox="0 0 12 10"
                    >
                      <polyline
                        points="1.5 6 4.5 9 10.5 1"
                      />
                    </svg>
                  </div>
                </span>
                <span
                  class="chakra-checkbox__label css-1oeb2oe"
                  data-checked=""
                >
                  <p
                    class="chakra-text css-vjer7o"
                  >
                    A Better Option
                  </p>
                </span>
              </label>
              <div
                class="chakra-collapse"
                style="overflow: initial; display: block; opacity: 1; height: auto;"
              >
                <div
                  class="css-1s47xlp"
                >
                  <div
                    class="chakra-form-control css-0"
                    role="group"
                  >
                    <label
                      class="chakra-form__label css-uvho5g"
                      data-cy="Describe the data source:"
                      for="field-9"
                      id="field-9-label"
                    >
                      Describe the data source:
                    </label>
                    <textarea
                      class="chakra-textarea css-0"
                      data-cy="DataSourceSelections.ABetterOption.description"
                      id="field-9"
                      name="DataSourceSelections.ABetterOption.description"
                    >
                      dun dun dun
                    </textarea>
                    <div
                      class="css-k008qs"
                    >
                      <div
                        class="css-17xejub"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="css-79elbk"
            >
              <label
                class="chakra-checkbox css-1uiwwan"
                data-checked=""
                data-cy="DataSource2"
              >
                <input
                  checked=""
                  class="chakra-checkbox__input"
                  id="DataSource2-checkbox"
                  style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
                  type="checkbox"
                  value="SomethingEvenCrazierThanTheFirst2"
                />
                <span
                  aria-hidden="true"
                  class="chakra-checkbox__control css-dnty2r"
                  data-checked=""
                >
                  <div
                    style="display: flex; align-items: center; justify-content: center; height: 100%; transform: none;"
                  >
                    <svg
                      class="css-o8vbdw"
                      style="fill: none; stroke-width: 2; stroke: currentColor; stroke-dasharray: 16; opacity: 1; stroke-dashoffset: 0;"
                      viewBox="0 0 12 10"
                    >
                      <polyline
                        points="1.5 6 4.5 9 10.5 1"
                      />
                    </svg>
                  </div>
                </span>
                <span
                  class="chakra-checkbox__label css-1oeb2oe"
                  data-checked=""
                >
                  <p
                    class="chakra-text css-vjer7o"
                  >
                    Something Even Crazier Than The First 2
                  </p>
                </span>
              </label>
              <div
                class="chakra-collapse"
                style="overflow: initial; display: block; opacity: 1; height: auto;"
              >
                <div
                  class="css-1ysiprv"
                />
              </div>
            </div>
          </div>
          <div
            class="css-k008qs"
          >
            <div
              class="css-17xejub"
            />
          </div>
        </div>
      </div>
      <div
        class="chakra-stack css-owjkmg"
      >
        <p
          class="chakra-text css-t3k68f"
        >
          For each data source selected above, describe which reporting entities used each data source (e.g., health plans, FFS). If the data source differed across health plans or delivery systems, identify the number of plans that used each data source:
        </p>
        <div
          class="chakra-form-control css-0"
          role="group"
        >
          <label
            class="chakra-form__label css-uvho5g"
            data-cy="Data Source Description"
            for="field-10"
            id="field-10-label"
          >
            Data Source Description
          </label>
          <textarea
            class="chakra-textarea css-0"
            data-cy="DataSourceDescription"
            id="field-10"
            name="DataSourceDescription"
          >
            Another one bites the dust
          </textarea>
          <div
            class="css-k008qs"
          >
            <div
              class="css-17xejub"
            />
          </div>
        </div>
      </div>
    </fieldset>
  </div>
</body>
`;

exports[`Test the global DataSource component (Default) Component renders with correct content 1`] = `
<body>
  <div
    class="css-pb69ky"
  >
    <h2
      class="chakra-form__label css-e3y4kl"
      data-cy="Data Source"
    >
      Data Source
    </h2>
    <fieldset>
      <div
        data-cy="data-source-options"
      >
        <div
          class="chakra-form-control css-0"
          role="group"
        >
          <label
            class="chakra-form__label css-uvho5g"
            data-cy="If reporting entities (e.g., health plans) used different data sources, please select all applicable data sources used below."
            for="field-1"
            id="field-1-label"
          >
            If reporting entities (e.g., health plans) used different data sources, please select all applicable data sources used below.
          </label>
          <div
            class="chakra-stack css-n21gh5"
          >
            <div
              class="css-79elbk"
            >
              <label
                class="chakra-checkbox css-1uiwwan"
                data-checked=""
                data-cy="DataSource0"
              >
                <input
                  checked=""
                  class="chakra-checkbox__input"
                  id="DataSource0-checkbox"
                  style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
                  type="checkbox"
                  value="AdministrativeData"
                />
                <span
                  aria-hidden="true"
                  class="chakra-checkbox__control css-dnty2r"
                  data-checked=""
                >
                  <div
                    style="display: flex; align-items: center; justify-content: center; height: 100%; transform: none;"
                  >
                    <svg
                      class="css-o8vbdw"
                      style="fill: none; stroke-width: 2; stroke: currentColor; stroke-dasharray: 16; opacity: 1; stroke-dashoffset: 0;"
                      viewBox="0 0 12 10"
                    >
                      <polyline
                        points="1.5 6 4.5 9 10.5 1"
                      />
                    </svg>
                  </div>
                </span>
                <span
                  class="chakra-checkbox__label css-1oeb2oe"
                  data-checked=""
                >
                  <p
                    class="chakra-text css-vjer7o"
                  >
                    Administrative Data
                  </p>
                </span>
              </label>
              <div
                class="chakra-collapse"
                style="overflow: initial; display: block; opacity: 1; height: auto;"
              >
                <div
                  class="css-1s47xlp"
                >
                  <div
                    class="chakra-form-control css-0"
                    role="group"
                  >
                    <label
                      class="chakra-form__label css-uvho5g"
                      data-cy="What is the Administrative Data Source?"
                      for="field-2"
                      id="field-2-label"
                    >
                      What is the Administrative Data Source?
                    </label>
                    <div
                      class="chakra-stack css-n21gh5"
                    >
                      <div
                        class="css-79elbk"
                      >
                        <label
                          class="chakra-checkbox css-1uiwwan"
                          data-checked=""
                          data-cy="DataSourceSelections.AdministrativeData0.selected0"
                        >
                          <input
                            checked=""
                            class="chakra-checkbox__input"
                            id="DataSourceSelections.AdministrativeData0.selected0-checkbox"
                            style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
                            type="checkbox"
                            value="MedicaidManagementInformationSystemMMIS"
                          />
                          <span
                            aria-hidden="true"
                            class="chakra-checkbox__control css-dnty2r"
                            data-checked=""
                          >
                            <div
                              style="display: flex; align-items: center; justify-content: center; height: 100%; transform: none;"
                            >
                              <svg
                                class="css-o8vbdw"
                                style="fill: none; stroke-width: 2; stroke: currentColor; stroke-dasharray: 16; opacity: 1; stroke-dashoffset: 0;"
                                viewBox="0 0 12 10"
                              >
                                <polyline
                                  points="1.5 6 4.5 9 10.5 1"
                                />
                              </svg>
                            </div>
                          </span>
                          <span
                            class="chakra-checkbox__label css-1oeb2oe"
                            data-checked=""
                          >
                            <p
                              class="chakra-text css-vjer7o"
                            >
                              Medicaid Management Information System (MMIS)
                            </p>
                          </span>
                        </label>
                        <div
                          class="chakra-collapse"
                          style="overflow: initial; display: block; opacity: 1; height: auto;"
                        >
                          <div
                            class="css-1ysiprv"
                          />
                        </div>
                      </div>
                      <div
                        class="css-79elbk"
                      >
                        <label
                          class="chakra-checkbox css-1uiwwan"
                          data-checked=""
                          data-cy="DataSourceSelections.AdministrativeData0.selected1"
                        >
                          <input
                            checked=""
                            class="chakra-checkbox__input"
                            id="DataSourceSelections.AdministrativeData0.selected1-checkbox"
                            style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
                            type="checkbox"
                            value="AdministrativeDataOther"
                          />
                          <span
                            aria-hidden="true"
                            class="chakra-checkbox__control css-dnty2r"
                            data-checked=""
                          >
                            <div
                              style="display: flex; align-items: center; justify-content: center; height: 100%; transform: none;"
                            >
                              <svg
                                class="css-o8vbdw"
                                style="fill: none; stroke-width: 2; stroke: currentColor; stroke-dasharray: 16; opacity: 1; stroke-dashoffset: 0;"
                                viewBox="0 0 12 10"
                              >
                                <polyline
                                  points="1.5 6 4.5 9 10.5 1"
                                />
                              </svg>
                            </div>
                          </span>
                          <span
                            class="chakra-checkbox__label css-1oeb2oe"
                            data-checked=""
                          >
                            <p
                              class="chakra-text css-vjer7o"
                            >
                              Administrative Data Other
                            </p>
                          </span>
                        </label>
                        <div
                          class="chakra-collapse"
                          style="overflow: initial; display: block; opacity: 1; height: auto;"
                        >
                          <div
                            class="css-1s47xlp"
                          >
                            <div
                              class="chakra-form-control css-0"
                              role="group"
                            >
                              <label
                                class="chakra-form__label css-uvho5g"
                                data-cy="Describe the data source:"
                                for="field-3"
                                id="field-3-label"
                              >
                                Describe the data source:
                              </label>
                              <textarea
                                class="chakra-textarea css-0"
                                data-cy="DataSourceSelections.AdministrativeData0-AdministrativeDataOther.description"
                                id="field-3"
                                name="DataSourceSelections.AdministrativeData0-AdministrativeDataOther.description"
                              >
                                Another one bites the dust
                              </textarea>
                              <div
                                class="css-k008qs"
                              >
                                <div
                                  class="css-17xejub"
                                />
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                    <div
                      class="css-k008qs"
                    >
                      <div
                        class="css-17xejub"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="css-79elbk"
            >
              <label
                class="chakra-checkbox css-1uiwwan"
                data-checked=""
                data-cy="DataSource1"
              >
                <input
                  checked=""
                  class="chakra-checkbox__input"
                  id="DataSource1-checkbox"
                  style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
                  type="checkbox"
                  value="OtherDataSource"
                />
                <span
                  aria-hidden="true"
                  class="chakra-checkbox__control css-dnty2r"
                  data-checked=""
                >
                  <div
                    style="display: flex; align-items: center; justify-content: center; height: 100%; transform: none;"
                  >
                    <svg
                      class="css-o8vbdw"
                      style="fill: none; stroke-width: 2; stroke: currentColor; stroke-dasharray: 16; opacity: 1; stroke-dashoffset: 0;"
                      viewBox="0 0 12 10"
                    >
                      <polyline
                        points="1.5 6 4.5 9 10.5 1"
                      />
                    </svg>
                  </div>
                </span>
                <span
                  class="chakra-checkbox__label css-1oeb2oe"
                  data-checked=""
                >
                  <p
                    class="chakra-text css-vjer7o"
                  >
                    Other Data Source
                  </p>
                </span>
              </label>
              <div
                class="chakra-collapse"
                style="overflow: initial; display: block; opacity: 1; height: auto;"
              >
                <div
                  class="css-1s47xlp"
                >
                  <div
                    class="chakra-form-control css-0"
                    role="group"
                  >
                    <label
                      class="chakra-form__label css-uvho5g"
                      data-cy="Describe the data source:"
                      for="field-4"
                      id="field-4-label"
                    >
                      Describe the data source:
                    </label>
                    <textarea
                      class="chakra-textarea css-0"
                      data-cy="DataSourceSelections.OtherDataSource.description"
                      id="field-4"
                      name="DataSourceSelections.OtherDataSource.description"
                    >
                      dun dun dun
                    </textarea>
                    <div
                      class="css-k008qs"
                    >
                      <div
                        class="css-17xejub"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="css-k008qs"
          >
            <div
              class="css-17xejub"
            />
          </div>
        </div>
      </div>
      <div
        class="chakra-stack css-owjkmg"
      >
        <p
          class="chakra-text css-t3k68f"
        >
          For each data source selected above, describe which reporting entities used each data source (e.g., health plans, FFS). If the data source differed across health plans or delivery systems, identify the number of plans that used each data source:
        </p>
        <div
          class="chakra-form-control css-0"
          role="group"
        >
          <label
            class="chakra-form__label css-uvho5g"
            data-cy="Data Source Description"
            for="field-5"
            id="field-5-label"
          >
            Data Source Description
          </label>
          <textarea
            class="chakra-textarea css-0"
            data-cy="DataSourceDescription"
            id="field-5"
            name="DataSourceDescription"
          >
            Another one bites the dust
          </textarea>
          <div
            class="css-k008qs"
          >
            <div
              class="css-17xejub"
            />
          </div>
        </div>
      </div>
    </fieldset>
  </div>
</body>
`;
